client/src/main.js

```diff
@@ -26,5 +26,6 @@ if (token) {
 
 new Vue({
   router,
+  store,
```

Comment:
This is the first change I described. The dispatch error pops up because Vue doesn't yet have a store, so the Login component is calling dispatch on the undefined field.